### PR TITLE
281 document history view 2

### PR DESCRIPTION
**Pull request by clemens-msupply**
Adds a basic history view modal to the patient detail page. Shows a list of edits and shows json diffs when selecting an edit.  At this stage its mainly a dev feature and not yet planned to be included for V1. Think its a nice feature to show that we have a document history though and possibly helpful for debugging.

@mark-prins should I move it to common/src/ui/components? (should the ui/forms folder go to ui/components as well?)

#284

### CODE REVIEW

File: client/packages/system/src/Patient/DocumentHistory/DocumentHistory.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import { DocumentNode } from 'packages/common/src/types/schema';
+import React, { FC, useEffect, useState } from 'react';
+import ReactDiffViewer from 'react-diff-viewer';
+import { usePatient } from '../api';
+import { Box } from 'packages/common/src';
+
+export const DocumentHistory: FC<{ documentName: string }> = ({
+  documentName,
+}) => {
+  const [history, setHistory] = useState([] as DocumentNode[]);
```
Comment (mark-prins) [inline client/packages/system/src/Patient/DocumentHistory/DocumentHistory.tsx:17]: I don't see the need for local state here.

I would remove the `useEffect` and change the hook call to be 
```  
const { data: docHistory, isLoading } = usePatient.document.history(documentName);
```

and add

```
  if (isLoading) return <BasicSpinner />;

```

and `(docHistory ?? []).map((h, i)` to make typescript happier ( also change the diff viewer usage ).

File: client/packages/system/src/Patient/DocumentHistory/DocumentHistory.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  };
+
+  return (
+    <div>
+      {history.map((h, i) => (
+        <div key={i}>
+          <Accordion
+            key={i}
+            sx={{
+              mt: '0 !important',
```
Comment (mark-prins) [inline client/packages/system/src/Patient/DocumentHistory/DocumentHistory.tsx:52]: I didn't see any effect from all of these margin changes. are they necessary?

File: client/packages/system/src/Patient/DocumentHistory/DocumentHistory.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                  Edit {history.length - i}
+                </Typography>
+                <Typography
+                  sx={{
+                    fontWeight: 'normal',
+                    textAlign: 'start',
+                    whiteSpace: 'nowrap',
+                  }}
+                >
+                  {h.timestamp}
```
Comment (mark-prins) [inline client/packages/system/src/Patient/DocumentHistory/DocumentHistory.tsx:84]: You could maybe use one of the formatters in DateUtils for this

**Review by mark-prins — APPROVED** [review]
👍  That looks super useful!

Like the use of the hook and modal - nice separation of the modal components into their own file.
I wouldn't add to common at this stage, as the diff component is fairly specific to this usage - but we may add if we find another use for it

File: client/packages/system/src/Patient/DetailView/AddBarButtons.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      >
+        <DetailContainer>
+          <Box
+            display="flex"
+            flexDirection="column"
+            alignItems="center"
+            gap={2}
+          >
+            <Typography sx={{ fontSize: 18, fontWeight: 700 }}>
+              Document Edit History
```
[the remark this answers is not in the document]
Comment (clemens-msupply) [reply client/packages/system/src/Patient/DetailView/AddBarButtons.tsx:52]: Ok I leave it for now and can change if we include it

File: client/packages/system/src/Patient/DocumentHistory/DocumentHistory.tsx (hunk 1 above)
Comment (clemens-msupply) [reply client/packages/system/src/Patient/DocumentHistory/DocumentHistory.tsx]: thanks done, I just check if the return value is not undefined...

**Comment by clemens-msupply** [comment]
Thanks for the feedback!

**Pull request merged**